app/controllers/courses_controller.rb

```diff
@@ -144,6 +144,29 @@ def favorite
     end
   end
 
+  def flag
+    @lesson = params[:lesson_id].present? ? Lesson.find(params[:lesson_id]) : nil
+    hash = {flagable_type: "Course", flagable_id: @course.id, user_id: current_user.id }
+    @flag = Flag.new(hash)
+    if @flag.save
+      if flag_params[:source] == "course_edit"
```

Comment:
A user can only get to the edit page for their own material so they would not be flagging it themselves.  We only need to flag from the show page.